app/notification.go

```diff
@@ -534,12 +525,53 @@ type ExplicitMentions struct {
 	ChannelMentioned bool
 }
 
+const (
+	// Different types of mentions ordered by their priority from lowest to highest
+
+	// The post is in a thread that the user has commented on
+	ThreadMention MentionType = iota
+
+	// The post is a comment on a thread started by the user
+	CommentMention
+
+	// The post contains an at-channel, at-all, or at-here
+	ChannelMention
+
+	// The post is a DM
+	DMMention
+
+	// The post contains an at-mention for the user
+	KeywordMention
+)
+
+type MentionType int
+
+func (m *ExplicitMentions) addMention(userId string, mentionType MentionType) {
+	if m.Mentions == nil {
+		m.Mentions = make(map[string]MentionType)
+	}
+
+	if currentType, ok := m.Mentions[userId]; ok && currentType >= mentionType {
+		return
+	}
+
+	m.Mentions[userId] = mentionType
+}
+
+func (m *ExplicitMentions) addMentions(userIds []string, mentionType MentionType) {
+	for _, userId := range userIds {
+		m.addMention(userId, mentionType)
+	}
+}
+
+func (m *ExplicitMentions) removeMention(userId string) {
+	delete(m.Mentions, userId)
```

Comment:
is it possible to have two mentions, overwrite one with the other and then delete? Since we are going per-post I think it would be reaaaally hard. but I thought I'd raise the question.